core/src/main/java/org/apache/accumulo/core/client/AccumuloClient.java

```diff
@@ -276,6 +294,14 @@ ConditionalWriter createConditionalWriter(String tableName, ConditionalWriterCon
   AccumuloClient changeUser(String principal, AuthenticationToken token)
       throws AccumuloSecurityException, AccumuloException;
 
+  /**
+   * Cleans up any resources created by an AccumuloClient like threads and sockets. Anything created
+   * from this client will likely not work after calling this method. For example a Scanner created
+   * using this client will likely fail after close is called.
+   */
+  @Override
+  public void close();
```

Comment:
Currently I can not think of a use case for throwing a checked exception.  The code will attempt resource cleanup and should only fail if there is a bug in the code, in which case a runtime exception is fine.

For close methods that wait, the user definitely wants to know if there is a problem and a checked exception would make sense in this case.  However, the expected behavior of this API is not to wait for current operations to complete. Instead of waiting, calling close causes any current or future operations to fail.  This behavior does not seem to need a checked exception.

I am still a bit uncertain about this. With exceptions in APIs, we are stuck with whatever is decided now.  Can not add or remove checked exceptions to API methods w/o breaking user code.

Reply:
For options 1, may be able to do the following in the future for a specific error.
 * Create class X that extends RuntimeException for the error.
 * Unsure about this, but may be able to add `throws X` to the method.  This wold basically be done for documentation purposes.
 * Document that X is thrown in the javadoc and why its thrown.

For options 3.a and 3.b we would be unable to say why the exceptions are thrown in javadoc.